3rdparty/Makefile

```diff
@@ -1,3 +1,14 @@
+#
+# Note that this Makefile is borked and it will rebuild everything every time you run it. This is due
+# me depending on $(BUILD_DIR) everywhere.
```

Comment:
Well, it will always think that every target that depens on build folder must be rebuilt. So it build everything correctly and not errors, it just does it too every time you hit make.

Currently the target was to make it possible to build once on travis so I didn't bother to fix.